### ISSUE TITLE
Import game folder structure from console

### ISSUE THREAD

**Issue by yepyepyep4711** [TeamShinkansen/Hakchi2-CE#240]
**Is your feature request related to a problem? Please describe.**
I usually make changes to the console from my laptop. Now I don't have access to it. I my case, it's temporary, but imagine it would be lost or broken. I want to make changes to the console (add two games) without losing the directory structure or anything really. Just add two games to the existing structure. Sadly, if i connect the console to hakchi on another machine, I don't see a possibility to import the structure, edit it and sync again.

**Describe the solution you'd like**
Tools (or File) > "Import games, art, menus and settings from console" (probably needs to be shorter, like "sync from console"), with subsequent correct display of the game list, setting of the "structure" menu, etc.

**Describe alternatives you've considered**
None comes to mind.

**Comment by Xeddius** [comment]
This is a much needed feature as I'm currently in EXACTLY that situation. This should definitely already be a feature. I'm stuck with the option to either toss 4 new games on and lose 60, or just leave it as it is permanently until the devs come up with a reasonable solution.

**Comment by yepyepyep4711** [comment]
Hello, I just wanted to ask @KMFDManic or @DanTheMan827 if there could be any feedback on this. I've just hit this snag again. My Hakchi is installed, not portable, and set up for use with the console. I have a USB stick for additional games I set up some time ago, I've forgotten how. I've just tried adding one game that is too big for the console to the USB stick. Since my Hakchi is set up for the use with console and the directory structure is different, I made sure to only check that one game before syncing to the USB stick. And I ended up adding the one game but deleting everything else from the stick.

So, question #1: are there any plans to implement "sync from consoles" or any workarounds to manually download from the console everything Hakchi needs to be aware of its setup?
As a corollary, question #2: is there a way to migrate from using the console only to using a USB stick? I my case, I could simplify my setup by having only one "place" were everything is, but I would need to migrate roms, saved states, directory structure, covers, and what not from the console to the stick. Can this be done?

Many thanks for your time.

**Comment by DanTheMan827** [comment]
It's not exactly what you were describing, but 3.9.0 added a feature to import games from the console back to the computer.

It does not however import any of the folder structure or folder art.

**Comment by yepyepyep4711** [comment]
yup, saw that. It's great progress, thank you! At least there is a way of getting the games off the console Let's keep this open until we have the full functionality if you don't mind, though. Besides structure, things like core association and media would belong to "import catalogue from console", no?

I'm realising while I'm typing some people might understand "sync from console" to also include console settings and saved states and whatnot. Maybe a choosing dialogue is needed if that becomes available. Just thinking out lout here.
